lib/dal-test/src/helpers/change_set.rs

```diff
@@ -0,0 +1,122 @@
+        ctx.update_snapshot_to_visibility().await?;
+        Ok(())
+    }
+
+    /// Applies the current [`ChangeSet`] to its base [`ChangeSet`]. Then, it updates the snapshot
+    /// to the visibility without using an editing [`ChangeSet`]. In other words, the resulting,
+    /// snapshot is "HEAD" without an editing [`ChangeSet`].
+    pub async fn apply_change_set_to_base(ctx: &mut DalContext) -> ChangeSetTestHelpersResult<()> {
+        let applied_change_set = ChangeSet::apply_to_base_change_set(ctx, true).await?;
+
+        Self::commit_and_error_on_conflicts(ctx).await?;
+
+        ctx.update_visibility_and_snapshot_to_visibility_no_editing_change_set(
+            applied_change_set.base_change_set_id.ok_or(
+                ChangeSetTestHelpersError::BaseChangeSetNotFound(applied_change_set.id),
+            )?,
+        )
+        .await?;
+        Ok(())
+    }
+
+    /// Abandons the current [`ChangeSet`].
+    pub async fn abandon_change_set(ctx: &mut DalContext) -> ChangeSetTestHelpersResult<()> {
+        let mut abandonment_change_set = ChangeSet::find(ctx, ctx.change_set_id()).await?.ok_or(
+            ChangeSetTestHelpersError::ChangeSetNotFound(ctx.change_set_id()),
+        )?;
+        abandonment_change_set.abandon(ctx).await?;
+        Ok(())
+    }
+
+    /// "Forks" from the "HEAD" [`ChangeSet`], which is the default [`ChangeSet`] of the workspace.
+    /// The name of the forked [`ChangeSet`] will be random.
+    ///
+    /// If you'd like to provide a name, use [`Self::fork_from_head_change_set_with_name`].
+    pub async fn fork_from_head_change_set(ctx: &mut DalContext) -> ChangeSetTestHelpersResult<()> {
+        Self::fork_from_head_change_set_inner(ctx, generate_fake_name()).await
+    }
+
+    /// "Forks" from the "HEAD" [`ChangeSet`], which is the default [`ChangeSet`] of the workspace.
+    /// The name of the forked [`ChangeSet`] comes from the corresponding function parameter.
+    ///
+    /// If you'd like a randomly generated name, use [`Self::fork_from_head_change_set`].
+    pub async fn fork_from_head_change_set_with_name(
+        ctx: &mut DalContext,
+        name: impl AsRef<str>,
+    ) -> ChangeSetTestHelpersResult<()> {
+        Self::fork_from_head_change_set_inner(ctx, name).await
+    }
+
+    async fn fork_from_head_change_set_inner(
+        ctx: &mut DalContext,
+        name: impl AsRef<str>,
+    ) -> ChangeSetTestHelpersResult<()> {
+        let new_change_set = ChangeSet::fork_head(ctx, name).await?;
+
+        ctx.update_visibility_and_snapshot_to_visibility(new_change_set.id)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn commit_and_error_on_conflicts(ctx: &DalContext) -> ChangeSetTestHelpersResult<()> {
+        match ctx.blocking_commit().await? {
+            Some(conflicts) => Err(ChangeSetTestHelpersError::ConflictsFoundAfterCommit(
+                conflicts,
+            )),
+            None => Ok(()),
+        }
+    }
```

Comment:
Conflict errors will now bubble up to the location in the test. You'll have more context on an error log rather than an inner-panic in a helper function.